### DISCUSSION TITLE
Ignore duplicates missing

### DISCUSSION THREAD

**Discussion by KyleSanderson** [autobrr/autobrr#294] [Ideas]
Now that the index is searchable (nice!) - hopefully the ignore duplicates flag from autodl-irssi becomes a thing as there are much better ways to handle this now.

**Comment by ludviglundgren** [comment]
I converted this to a GH Discussion for now. We have talked a lot about this in Discord in the past but information and discussions easily gets lost.

Lets start to define what a duplicate is since it can be many things.

Examples:
 - Full release name
 - Show/game/title
 - Combo of title, resolution, group etc
 - Same release on multiple indexers
 - Season and episode

Should it be static or configurable? A basic solution would be faster to develop but might not be what users are looking for.

All feedback are welcome :smiley:

**Reply by blixten85** [reply]
How is the integration of this with the other *arrs? And notifiarrs news profile?

in general the more customisable it is the better, but that depends on wetter we have to regexp or can just tick boxes in a drop list. I prefer the easiest choice ofc.

**Comment by KyleSanderson** [comment]
Ultimately, what constitutes a dupe is if the same content is there. So improving upon what autodl-irssi had would be if the content layout is the same as the existing file (or, converted to match the same layout, either modifying new or existing depending on what's possible). If there are additional files on the new submission (Sample/Srt as an example) that should still count as a duplicate. Parsing the title for Title / Season / Year / Episode should be considered hints, but as some folks retag the exact same content it's not required. Piece size should be ignored.

**Reply by KyleSanderson** [reply]
Specifically from qBittorrent this is exposed through the files API as far as I'm aware.

**Comment by LSL1337** [comment]
I think short term solution could be that if the izem has the same name, it should be considered as duplicate.
Maybe it can be added with regardless of resolution.
If i download episode1 1080p, other episode 1 releases can be ignored.
More logic would make it much harder to implememt, and it would still be short of sonarr/radarr. But this could cover a lot of use cases, which are not compatible with sonarr/radarr. (Sports for example)

**Comment by luckylittle** [comment]
It has been a while since this discussion started - can we do something really simple to start with? This is a number one feature that I am really missing. Simple to me is: if the release name **exactly** matches and the push on the first one was successfully approved, then ignore all of the sub-sequential pushes. I don't need to wait for it to complete - some 0day trackers can push the same release within a few seconds from each other.

Pseudo code was mentioned previously here: https://github.com/autobrr/autobrr/issues/375#issuecomment-1254589644

**Reply by KyleSanderson** [reply]
Upgraderr has existed for years, and is documented on autobrr.com

**Comment by rpatterson** [comment]
Isn't this implemented by [duplicate profiles](https://github.com/autobrr/autobrr/pull/1711)?

**Comment by zze0s** [comment]
Implemented as of [this PR](https://github.com/autobrr/autobrr/pull/1711) in version [v1.56.0](https://github.com/autobrr/autobrr/releases/tag/v1.56.0) from December 2024.
